From e95bbe96b83047700e5dc6b5849f261157b88f04 Mon Sep 17 00:00:00 2001
From: "C, Udhayakumar" <udhayakumar.c@intel.com>
Date: Fri, 22 Jan 2021 19:28:16 +0530
Subject: [PATCH 024/109] 
 FIX-NULL-ADAPOTER-AND-REMOVING-CRITICAL-TRIP-POINT-FROM-IA

Signed-off-by: C, Udhayakumar <udhayakumar.c@intel.com>
---
 drivers/misc/intel_tsens/intel_tsens_host.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/misc/intel_tsens/intel_tsens_host.c b/drivers/misc/intel_tsens/intel_tsens_host.c
index adb553f3f2e3..e76a3a10fefe 100644
--- a/drivers/misc/intel_tsens/intel_tsens_host.c
+++ b/drivers/misc/intel_tsens/intel_tsens_host.c
@@ -118,7 +118,8 @@ static int intel_tsens_thermal_get_trip_type(struct thermal_zone_device *zone,
 	struct intel_tsens_host *tsens =
 		(struct intel_tsens_host *)zone->devdata;
 
-	*type = tsens->trip_info[trip]->trip_type;
+	if (tsens->trip_info[trip]->trip_type != THERMAL_TRIP_CRITICAL)
+		*type = tsens->trip_info[trip]->trip_type;
 	return 0;
 }
 
@@ -128,7 +129,8 @@ static int intel_tsens_thermal_get_trip_temp(struct thermal_zone_device *zone,
 	struct intel_tsens_host *tsens =
 		(struct intel_tsens_host *)zone->devdata;
 
-	*temp = tsens->trip_info[trip]->temp;
+	if (tsens->trip_info[trip]->trip_type != THERMAL_TRIP_CRITICAL)
+		*temp = tsens->trip_info[trip]->temp;
 	return 0;
 }
 
-- 
2.25.1

